### PR TITLE
add json editor component

### PR DESCRIPTION
**Pull request by jmhauck**

### CODE REVIEW

**Comment by jmhauck** [comment]
@MikeTschudi code scan errors with some of the Ace files...thoughts?

File: src/components/json-editor/json-editor.tsx [hunk 1]
```diff
@@ -0,0 +1,386 @@
...
+ *
+ * Attributes:
+ * `id`: Id of component (required, because it's used to distinguish between multiple instances of component)
+ * `content`: Initial content of editor
+ * `start-editing`: Hover & a11y text for the start-editing button; default is "Start editing"
+ * `search`: Hover & a11y text for the search button; default is "Search"
+ * `cancel-edits`: Hover & a11y text for the cancel-edits button; default is "Cancel edits"
+ * `save-edits`: Hover & a11y text for the save-edits button; default is "Save edits"
+ *
+ * @example
```
Comment (MikeTschudi) [inline src/components/json-editor/json-editor.tsx:29]: This example doesn't seem to match the code

**Review by MikeTschudi — DISMISSED** [review]

**Comment by MikeTschudi** [comment]
CodeQL is flagging this part of ace/worker-json.js:
image

Let's please go ahead with merge after [example is fixed](https://github.com/Esri/solutions-components/pull/5#discussion_r593343396) while investigating if anything can be done in worker-json.js.

File: index.html [hunk 2]
```diff
@@ -32,6 +32,8 @@
 
   <script type="module" src="https://unpkg.com/@esri/calcite-components/dist/calcite/calcite.esm.js"></script>
   <script type="module" src="dist/solutions/solutions.esm.js"></script>
+  <script src="../lib/ace/ace.js"></script>
```
Comment (MikeTschudi) [inline index.html:35]: On my system, these imports should be "./lib..."

**Comment by jmhauck** [comment]
I'll update the example again while putting the translations logic in place.

**Pull request merged**